lib/turbolinks/x_domain_blocker.rb

```diff
@@ -4,18 +4,18 @@ module Turbolinks
   # - The request is being redirected to a different domain
   module XDomainBlocker
     private
-      def same_origin?(a, b)
```

Comment:
@kaspth I don't like that we're defining a method with such a generic name in a module that's then included in `ActionController::Base`.

Also, the decision to return `false` on invalid URI is IMO specific to `XDomainBlocker` and may not match other applications' definition of what constitutes `same_origin?`.